### ISSUE TITLE
Monitoring V2 should prevent deploying unless enough CPU / Memory is available.

### ISSUE THREAD

**Issue by sangeethah** [rancher/dashboard#1849]
Rancher server version - v2.5.0-rc9

Steps that lead to the problem:
Create a cluster with 3 nodes (all role) + 1 worker node setup ( each node size --  2 GB Memory / 60 GB Disk /Ubuntu 18.04 (LTS) x64) 
Deploy v2 Monitoring
It failed to come up (partially deployed)
At this point , etcd on one of the nodes reported being in bad health.

Tried to delete Monitoring app (Which also failed)
Noticed that cattle-cluster-agent gets into a bad state - similar to https://github.com/rancher/rancher/issues/29176

Added more worker nodes
Deleted Monitoring app.
Deployed Monitoring app.
There was a complain about Alerts related secrets still existing , so cleared all the secrets from `cattle-monitoring-system`,
Deployed Monitoring app again.

This time I get the following error:

```
Release “rancher-monitoring-crd” does not exist. Installing it now.
checking 4 resources for changes
SUCCESS: helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-crd-9.4.200.yaml --version=9.4.200 --wait=true rancher-monitoring-crd /home/shell/helm/rancher-monitoring-crd-9.4.200.tgz
---------------------------------------------------------------------
helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-9.4.200.yaml --version=9.4.200 --wait=true rancher-monitoring /home/shell/helm/rancher-monitoring-9.4.200.tgz
Release “rancher-monitoring” does not exist. Installing it now.
Starting delete for “rancher-monitoring-admission” PodSecurityPolicy
podsecuritypolicies.policy “rancher-monitoring-admission” not found
creating 1 resource(s)
Starting delete for “rancher-monitoring-admission” ServiceAccount
serviceaccounts “rancher-monitoring-admission” not found
creating 1 resource(s)
Starting delete for “rancher-monitoring-admission” ClusterRole
clusterroles.rbac.authorization.k8s.io “rancher-monitoring-admission” not found
creating 1 resource(s)
Starting delete for “rancher-monitoring-admission” ClusterRoleBinding
clusterrolebindings.rbac.authorization.k8s.io “rancher-monitoring-admission” not found
creating 1 resource(s)
Starting delete for “rancher-monitoring-admission” Role
roles.rbac.authorization.k8s.io “rancher-monitoring-admission” not found
creating 1 resource(s)
Starting delete for “rancher-monitoring-admission” RoleBinding
rolebindings.rbac.authorization.k8s.io “rancher-monitoring-admission” not found
creating 1 resource(s)
Starting delete for “rancher-monitoring-admission-create” Job
jobs.batch “rancher-monitoring-admission-create” not found
creating 1 resource(s)
Watching for changes to Job rancher-monitoring-admission-create with timeout of 10m0s
Add/Modify event for rancher-monitoring-admission-create: MODIFIED
rancher-monitoring-admission-create: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
Add/Modify event for rancher-monitoring-admission-create: MODIFIED
creating 1 resource(s)
creating 1 resource(s)
creating 1 resource(s)
creating 1 resource(s)
creating 1 resource(s)
Error: failed pre-install: warning: Hook pre-install rancher-monitoring/templates/alertmanager/secret.yaml failed: Internal error occurred: resource quota evaluates timeout
```

We should be able to prevent the user from getting into a situation were control/etcd plane could go bad, There are some checks done for this in v1 monitoring , Should we consider doing similar  checks for v2 monitoring charts ?

**Comment by maggieliu** [comment]
Going forward: add annotations for CPU and Memory requirements on the charts. If those are not met then UI will show a warning but it won't prevent user from deploying the chart. 

This will apply to:
* Monitoring
* Istio
* RIO

**Comment by deniseschannon** [comment]
Monitoring chart is adding 
```
catalog.cattle.io/requests-cpu: "4500m"
catalog.cattle.io/requests-memory: "4000Mi"
```

**Comment by deniseschannon** [comment]
UI needs to disable the launch button for the app deployment if a chart has these labels and there isn't enough resources to deploy.

**Comment by gaktive** [comment]
There's similar logic in Ember already but not directly applicable.

**Comment by gaktive** [comment]
QA has asked for an update on this.

**Comment by nwmac** [comment]
@vincent99 Is this something you will get done for 2.5.6? or shall we bump to 2.5.7?

**Comment by vincent99** [comment]
Yes, https://github.com/rancher/dashboard/pull/2366 .

**Comment by jiaqiluo** [comment]
The bug fix is validated on v2.5-head a85f4d1

When the downstream cluster has available less than 4.5 CPU cores, the following page show when you try to install the monitoring v2 on the dashboard. 

![image](https://user-images.githubusercontent.com/6218999/108767307-84eec100-7513-11eb-8dfc-926119f422dd.png)

**Comment by jiaqiluo** [comment]
Reopen this issue because I cannot edit/upgrade the monitoring app if the cluster does not have enough CPU left after installing the monitoring app.

**Comment by jiaqiluo** [comment]
the bug fix is validated on v2.5-head 87006de with the latest UI

It is confirmed that the check for available CPU does not happen when editing /upgrading the monitoring app.
